src/compiler.ts

```diff
@@ -9217,12 +9280,394 @@ export class Compiler extends DiagnosticEmitter {
     return module.block(label, conditions, NativeType.I32);
   }
 
+  checkInterfaceImplementation(toType: Type, fromType: Type, reportNode: Node): bool {
+    const _interface = (<Interface>toType.classReference!);
+    const _class = fromType.classReference!;
+    var imems: Map<string, Element>;
+    var mems: Map<string, Element> | null;
+    if (_interface.members == null) {
+      return true;
+    }
+    imems = _interface.members;
+    mems = _class.members;
+
+    var error = false;
+    var incorrectMember = false;
+    for (let [name, imem] of imems.entries()) {
+      error = error || incorrectMember;
+      if (!imem.is(CommonFlags.VIRTUAL)) {
+        continue;
+      }
+      incorrectMember = true;
+      if (mems == null || mems.get(name) == null) {
+        // Error!
+        this.error(
+          DiagnosticCode.Property_0_is_missing_in_type_1_but_required_in_type_2,
+          _class.identifierNode.range,
+          name,
+          _class.name,
+          _interface.name
+        );
+        continue;
+      }
+      let mem  = mems.get(name)!;
+      if (imem.kind != mem.kind) {
+        // Interfaces can't have properties
+        if (mem.kind != ElementKind.PROPERTY_PROTOTYPE
+            && mem.kind != ElementKind.FIELD_PROTOTYPE
+            && mem.kind != ElementKind.FIELD
+            && imem.kind == ElementKind.PROPERTY) {
+          this.error(
+            DiagnosticCode.Type_0_is_not_assignable_to_type_1,
+            (<DeclaredElement>mem).declaration.range,
+            mem.name,
+            name);
+          continue;
+        }
+        // Error
+      }
+      let from: Type = Type.void, to: Type = Type.void;
+      switch (mem.kind) {
+        case ElementKind.FIELD: {
+          mem = (<Field>mem).prototype;
+          imem = (<Property>imem).prototype;
+        }
+        case ElementKind.FIELD_PROTOTYPE: {
```

Comment:
Mixing prototypes and instances here seems odd, like if it's missing a distinction of static vs instance.